src/ringbuffer_trait.rs

```diff
@@ -222,14 +225,35 @@ mod iter {
 
         #[inline]
         fn next(&mut self) -> Option<Self::Item> {
-            if self.index < self.obj.len() {
+            if self.index < self.len {
                 let res = self.obj.get(self.index as isize);
                 self.index += 1;
                 res
             } else {
                 None
             }
         }
+
+        fn size_hint(&self) -> (usize, Option<usize>) {
+            (self.len, Some(self.len))
+        }
+    }
+
+    impl<'rb, T: 'rb, RB: RingBufferExt<T>> FusedIterator for RingBufferIterator<'rb, T, RB> { }
+
+    impl<'rb, T: 'rb, RB: RingBufferExt<T>> ExactSizeIterator for RingBufferIterator<'rb, T, RB> { }
+
+    impl<'rb, T: 'rb, RB: RingBufferExt<T>> DoubleEndedIterator for RingBufferIterator<'rb, T, RB> {
+        #[inline]
+        fn next_back(&mut self) -> Option<Self::Item> {
+            if self.len > 0 && self.index < self.len {
+                let res = self.obj.get((self.len - 1) as isize);
```

Comment:
this can be simplified to
```rust
let res = self.obj.get(-1)
```
Otherwise it looks good :+1:

Reply:
Are you certain? As far as I can tell, this would return the same element every time.

Reply:
Oh yeah you're right, my apologies.